### PR TITLE
Add instance_norm to paddle frontend

### LINKED ISSUE #19241: instance_norm

### PR DESCRIPTION
**Pull request by Madjid-CH**
Close #19241

### CODE REVIEW

**Comment by Madjid-CH** [comment]
the tests that I added are not working 
the error is a TypeError which says that "unexpected kword: offset" I added this kword to the function signature but it still not working

**Comment by Madjid-CH** [comment]
> 

sorry about that, I'll fix it right away

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Madjid-CH 

Thanks for the PR🙂

The function signature would be the same as the one indicated in the official framework documentation. Please refer here to correct the function signature and update the unit tests https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/nn/functional/instance_norm_en.html#instance-norm

Thanks😊

**Comment by hirwa-nshuti** [comment]
add_frontend_checklist

File: ivy/functional/frontends/paddle/nn/functional/norm.py [hunk 1]
```diff
@@ -1 +1,20 @@
...
+                  running_mean=None,
+                  running_var=None,
+                  weight=None,
+                  bias=None,
+                  use_input_stats=True,
+                  momentum=0.9,
+                  eps=1e-05,
+                  data_format='NCHW',
+                  name=None):
+    return ivy.instance_norm(x, running_mean, running_var,
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/paddle/nn/functional/norm.py:18]: `ivy.instance_norm` doesn't have training parameter 
You can refer to the documentation here
https://unify.ai/docs/ivy/docs/functional/ivy/experimental/norms/ivy.functional.ivy.experimental.norms.instance_norm.html#instance-norm

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Madjid-CH 

I tried to run your tests locally and failing I just highlighted the first error I noticed😊
It could be better if your run your tests before requesting a review and also there is no issue if the tests are failing feel free to request support but knowing what's going wrong.

Thanks 🙂

File: ivy/functional/frontends/paddle/nn/functional/norm.py (hunk 1 above)
Comment (Madjid-CH) [reply ivy/functional/frontends/paddle/nn/functional/norm.py:18]: sorry, I don't think I get what you mean
the training parameter is in the function signature 
image

Comment (Madjid-CH) [reply ivy/functional/frontends/paddle/nn/functional/norm.py:18]: sorry, I didn't quite get it, the ivy.instance_norm signature has "training" parameter
image

however  ivy.functional.backends.paddle.experimental.norms.instance_norm signature don't have that parameter.
I get this function to be executed by forcing the paddle backend
image

but it raises IvyNotImplementedException.

so please can you explain to me which function signature should I implement ?

**Comment by Madjid-CH** [comment]
> Hey @Madjid-CH
> 
> I tried to run your tests locally and failing I just highlighted the first error I noticed😊 It could be better if your run your tests before requesting a review and also there is no issue if the tests are failing feel free to request support but knowing what's going wrong.
> 
> Thanks 🙂

I tried to run the tests but I always encounter this error
ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_norm.py:72 (test_paddle_instance_norm[cpu-ivy.functional.backends.numpy-False-False])
args = (dtype('O'),), kwargs = {}

I search through the previous closed PR to find if someone stumbled upon a similar problem, but I didn't find something useful.

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_norm.py [hunk 2]
```diff
@@ -1,3 +1,99 @@
...
+        helpers.dtype_and_values(
+            dtype=x_dtype,
+            shape=shape1,
+            min_value=-1001,
+            max_value=999,
+            large_abs_safety_factor=24,
+            small_abs_safety_factor=24,
+            safety_factor_scale="log",
+        )
+    )
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_norm.py:48]:
```suggestion
    )

```

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Madjid-CH 

I'm unable to run the tests locally would you update your fork with latest master branch changes🙂
Thanks

**Comment by Madjid-CH** [comment]
> Hey @Madjid-CH
> 
> I'm unable to run the tests locally would you update your fork with latest master branch changes🙂 Thanks

certainly

File: ivy/functional/frontends/paddle/nn/functional/norm.py [hunk 3]
```diff
@@ -1 +1,25 @@
...
+                  running_mean,
+                  running_var,
+                  weight=None,
+                  bias=None,
+                  use_input_stats=True,
+                  momentum=0.9,
+                  eps=1e-05,
+                  data_format='NCHW',
+                  name=None):
+    return ivy.instance_norm(x,
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/paddle/nn/functional/norm.py:18]: The signature of `ivy.instance_norm` only has `x, mean, variance, offset, scale, training, eps, momentum, data_format, out` arguments all these extra arguments that you're adding are causing failures when calling the function🙂

**Review by hirwa-nshuti — COMMENTED** [review]
Hey @Madjid-CH

I hope that this comment will direct you to get your implementation done😊

File: ivy/functional/frontends/paddle/nn/functional/norm.py (hunk 3 above)
Comment (Madjid-CH) [reply ivy/functional/frontends/paddle/nn/functional/norm.py:18]: I did try this parameters in the beginning. But the tests keeps failing with this error:
ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_norm.py:73 (test_paddle_instance_norm[cpu-ivy.functional.backends.numpy-False-False])
args = (dtype('O'),), kwargs = {}

Comment (Madjid-CH) [reply ivy/functional/frontends/paddle/nn/functional/norm.py]: I can implement the instance_norm function in paddle backend to get rid of the  IvyNotImplementedException.
Can I do that ?

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey I spent some time doing some diggings and,
The issue on backend implementation is fixed now but there is a big difference in `ivy.instance_norm` and `paddle.instance_norm` I would advise you to implement this function with composition it is the way that can work for now we will explore to extend `ivy.instance_norm` to behave as this function.

**Comment by Madjid-CH** [comment]
> Hey I spent some time doing some diggings and, The issue on backend implementation is fixed now but there is a big difference in `ivy.instance_norm` and `paddle.instance_norm` I would advise you to implement this function with composition it is the way that can work for now we will explore to extend `ivy.instance_norm` to behave as this function.

so I have to implement the paddle.instance_norm function in the backend, is that right ?

**Comment by hirwa-nshuti** [comment]
> > Hey I spent some time doing some diggings and, The issue on backend implementation is fixed now but there is a big difference in `ivy.instance_norm` and `paddle.instance_norm` I would advise you to implement this function with composition it is the way that can work for now we will explore to extend `ivy.instance_norm` to behave as this function.
> 
> so I have to implement the paddle.instance_norm function in the backend, is that right ?

No you only need to implement it in the frontend using other ivy functions

**Comment by Madjid-CH** [comment]
> > > Hey I spent some time doing some diggings and, The issue on backend implementation is fixed now but there is a big difference in `ivy.instance_norm` and `paddle.instance_norm` I would advise you to implement this function with composition it is the way that can work for now we will explore to extend `ivy.instance_norm` to behave as this function.
> > 
> > 
> > so I have to implement the paddle.instance_norm function in the backend, is that right ?
> 
> No you only need to implement it in the frontend using other ivy functions

understood

**Comment by Madjid-CH** [comment]
Hey, @hirwa-nshuti  I am facing this Assertion Error in my implementation. Can you please guide me on how to resolve this.
image

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_norm.py [hunk 4]
```diff
@@ -1,15 +1,136 @@
...
+        input_dtypes=x_dtype,
+        x=x,
+        running_mean=running_mean,
+        running_var=running_var,
+        weight=weights,
+        bias=bias,
+        eps=eps,
+        momentum=momentum,
+        data_format=data_format,
+        use_input_stats=use_input_stats, )
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_norm.py:131]: This is duplicated

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey
Tests are failing locally, would you fix that and would you fix the lint errors you can install the pre-commit by using this guide https://unify.ai/docs/ivy/overview/contributing/setting_up.html#pre-commit

**Pull request closed without merging**